checkout-example-advanced/src/main/resources/static/adyenImplementation.js

```diff
@@ -1,105 +1,143 @@
+                            headers: {
+                                "Content-Type": "application/json",
+                            }
+                        }).then(response => response.json());
+
+                        if (!resultCode) {
+                            actions.reject();
+                        }
+
+                        actions.resolve({
+                            resultCode,
+                            action,
+                            order,
+                            donationToken
+                        });
+                    }
+                } catch (error) {
+                    console.error(error);
+                    actions.reject();
+                }
+            },
+            onPaymentCompleted: (result, component) => {
+                console.info("onPaymentCompleted", result, component);
+                handleOnPaymentCompleted(result, component);
+            },
+            onPaymentFailed: (result, component) => {
+                console.info("onPaymentFailed", result, component);
+                handleOnPaymentFailed(result, component);
+            },
+            onError: (error, component) => {
+                console.error("onError", error.name, error.message, error.stack, component);
+                window.location.href = "/result/error";
+            },
+            onAdditionalDetails: async (state, component) => {
+                console.info("onAdditionalDetails", state, component);
+                const response = await fetch("/api/payments/details", {
+                    method: "POST",
+                    body: state.data ? JSON.stringify(state.data) : "",
+                    headers: {
+                        "Content-Type": "application/json",
+                    }
+                }).then(response => response.json());
+
+                if (response.action) {
+                    component.handleAction(response.action);
+                } else {
+                    handleOnPaymentCompleted(response);
+                }
+            }
+        };
 
-// Calls your server endpoints
-async function callServer(url, data) {
-  const res = await fetch(url, {
-    method: "POST",
-    body: data ? JSON.stringify(data) : "",
-    headers: {
-      "Content-Type": "application/json",
-    },
-  });
+        const paymentMethodsConfiguration = {
+            card: {
+                showBrandIcon: true,
+                hasHolderName: true,
+                holderNameRequired: true,
+                name: "Credit or debit card",
+                amount: {
+                    value: 10000,
+                    currency: "EUR",
+                },
+                placeholders: {
+                    cardNumber: '1234 5678 9012 3456',
+                    expiryDate: 'MM/YY',
+                    securityCodeThreeDigits: '123',
+                    securityCodeFourDigits: '1234',
+                    holderName: 'Developer Relations Team'
```

Comment:
I think we should keep `J. Smith` as placeholder because of the E2E testing suite